kubernetes/template-generator/template_generator.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/env python
+
+import re
+import yaml
+import json
+
+from container_os import *
+
+### TEMPLATE GENERATORS
+class SanitizeException(Exception):
+    pass
+
+class TemplateGenerator(object):
+    """
+        @param string container_name    the name of migrated container
+        @param ContainerOS system       a os description module
+        @param list exposed_ports       a list of exposed ports
+    """
+    def __init__(self, container_name, system, exposed_ports = None, external_ips = None):
+        self.sanit_container_name = self.sanitize_container_name(container_name)
+        self.system = system
+        self.exposed_ports = exposed_ports
+        self.external_ips = external_ips
+
+
+    """
+        Generate a YAML template for kerberos service
+        @param Boolean yaml     if true, the output will be in yaml, else a
+                                dict will be returned
+
+
+        @return string          a template string in yaml format/dict
+    """
+    def generate_service_template(self, yaml=True):
+        service_template = {
+            "apiVersion": "v1",
+            "kind": "Service",
+            "metadata": {
+                "name": "leapp-{}-service".format(self.sanit_container_name),
+                "labels": {
+                    "name": "leapp-{}-label".format(self.sanit_container_name),
+                    }
+                },
+            "spec": {
+                "selector": {
+                    "name": "leapp-{}-label".format(self.sanit_container_name),
+                    },
+            }
+        }
+
+        if not self.external_ips is None:
```

Comment:
this is both eye poking and can cause unexpected results :-D use if is not rather the is not is None  https://stackoverflow.com/a/2710949